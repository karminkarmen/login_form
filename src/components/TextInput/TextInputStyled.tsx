import styled from 'styled-components';

type Props = {
    label: string, 
    type: string
};

const InputStyled = styled.input.attrs((props: Props) => ({
    type: props.type,
    name: props.label,
    id: props.label
}))<Props>`
    font-size: ${props => props.theme.fontSizes.medium};
    border-radius: ${props => props.theme.radius.medium};
    border: 2px solid ${props => props.theme.colors.main};
    padding: ${props => props.theme.paddings.medium};
    background: transparent;
    color: ${props => props.theme.colors.secondaryLightest};
    margin-bottom: 1rem;

    ::placeholder {
        color: ${props => props.theme.colors.background};
    }

    &:focus {
        outline: none;
        background: ${props => props.theme.colors.secondaryLighten};
        color: ${props => props.theme.colors.main};
    }
`;

export default InputStyled;