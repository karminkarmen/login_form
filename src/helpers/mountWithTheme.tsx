import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../configs/theme';

const mountWithTheme = (child: React.ReactElement) => {
    return mount(child, {
        wrappingComponent: ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    });
}

export default mountWithTheme;