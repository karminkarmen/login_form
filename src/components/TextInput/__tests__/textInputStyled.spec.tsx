import React from 'react';
import mountWithTheme from '../../../helpers/mountWithTheme';
import TextInputStyled from '../TextInputStyled';

describe('<TextInputStyled />', () => {

    const props: React.ComponentProps<typeof TextInputStyled> = {
        label: 'remember', 
        type: 'password'
    };

    it('matches snapshot', () => {
        const wrapper = mountWithTheme(<TextInputStyled {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should render label html tag with specified type, name and id properties', () => {
        const wrapper = mountWithTheme(<TextInputStyled {...props} />);

        expect(wrapper.find(`input[type='${props.type}']`)).toHaveLength(1);
        expect(wrapper.find(`input[name='${props.label}']`)).toHaveLength(1);
        expect(wrapper.find(`input[name='${props.label}']`)).toHaveLength(1);
    });

});
