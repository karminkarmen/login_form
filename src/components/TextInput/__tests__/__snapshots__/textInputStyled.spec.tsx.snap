// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInputStyled /> matches snapshot 1`] = `"<input label=\\"remember\\" type=\\"password\\" name=\\"remember\\" id=\\"remember\\" class=\\"sc-AxjAm hRdQeY\\">"`;
